Buildings/Controls/OBC/CDL/Discrete/DiscreteMovingMean.mo

```diff
@@ -0,0 +1,104 @@
+  Continuous.Gain gai(k=1/nSam) "Get the average"
+    annotation (Placement(transformation(extent={{60,-10},{80,10}})));
+
+equation
+  connect(u, reaRep.u) annotation (Line(points={{-120,0},{-62,0}},
+                 color={0,0,127}));
+  connect(reaRep.y, triSam.u) annotation (Line(points={{-39,0},{-22,0}}, color={0,0,127}));
+  connect(samTri.y, triSam.trigger) annotation (Line(points={{-39,-50},{-10,-50},
+          {-10,-11.8}}, color={255,0,255}));
+  connect(triSam.y, mulSum.u)  annotation (Line(points={{1,0},{18,0}}, color={0,0,127}));
+  connect(u, sampler.u) annotation (Line(points={{-120,0},{-80,0},{-80,50},{-22,
+          50}},
+        color={0,0,127}));
+  connect(mulSum.y, gai.u) annotation (Line(points={{41,0},{58,0}}, color={0,0,127}));
+  connect(gai.y, y) annotation (Line(points={{81,0},{110,0}}, color={0,0,127}));
+  annotation (
+  defaultComponentName="disMovMea",
+  Icon(coordinateSystem(preserveAspectRatio=true, extent={{-100,-100},{100,100}}),
+       graphics={
+       Rectangle(
+       extent={{-100,-100},{100,100}},
+       lineColor={0,0,127},
+       fillColor={223,211,169},
+       lineThickness=5.0,
+       borderPattern=BorderPattern.Raised,
+       fillPattern=FillPattern.Solid),
+       Text(
+       extent={{-150,150},{150,110}},
+       textString="%name",
+       lineColor={0,0,255}),
+        Line(points={{-88,0},{70,0}}, color={192,192,192}),
+        Polygon(
+          points={{92,0},{70,8},{70,-8},{92,0}},
+          lineColor={192,192,192},
+          fillColor={192,192,192},
+          fillPattern=FillPattern.Solid),
+        Line(points={{-78,68},{-78,-80}}, color={192,192,192}),
+        Polygon(
+          points={{-78,90},{-86,68},{-70,68},{-78,90}},
+          lineColor={192,192,192},
+          fillColor={192,192,192},
+          fillPattern=FillPattern.Solid),
+        Line(points={{-78,22},{-58,22},{-58,-22}}, color={28,108,200}),
+        Line(points={{-58,-22},{-42,-22},{-38,-22},{-38,22}}, color={28,108,200}),
+        Line(points={{-38,22},{-18,22},{-18,-22}}, color={28,108,200}),
+        Line(points={{-18,-22},{-2,-22},{2,-22},{2,22}}, color={28,108,200}),
+        Line(points={{2,22},{22,22},{22,-22}}, color={28,108,200}),
+        Line(points={{22,-22},{38,-22},{42,-22},{42,22}}, color={28,108,200}),
+        Line(points={{-78,10},{-58,10}}, color={217,67,180}),
+        Line(points={{-58,10},{-58,0}}, color={217,67,180}),
+        Line(points={{-58,0},{62,0}}, color={217,67,180}),
+        Line(points={{42,22},{62,22},{62,-22}}, color={28,108,200})}),
+Documentation(info="<html>
+<p>
+This block calculates the discrete moving mean values of an input signal.
+The continuous input signal is first sampled; then at each sampling rate, the 
+block outputs the average value of the past n samples including the current 
+sample. 
+</p>
+<p>
+During initialization, the initial value given by the user is the value for the 
```

Comment:
@kuzha : This should be changed. At the first sample, it should produce the input. At the next sample, it should produce the average of the past two samples, then the past three samples and so on up to `n` samples. Then, the block is indeed a discretized version of `CDL.Continuous.MovingMean`.